src/main/java/redis/clients/jedis/JedisSocketConfig.java

```diff
@@ -0,0 +1,22 @@
+package redis.clients.jedis;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLParameters;
+import javax.net.ssl.SSLSocketFactory;
+
+public interface JedisSocketConfig {
```

Comment:
Is this basically the socket factory without the host/port?

Reply:
Mostly, yes. I want to keep socket factory dedicated for creating socket and this one for holding config parameters.

Reply:
Does HostAndPortMapper make sense though for other types of sockets? (e.g. AFUNIX)

Reply:
YES. I am generalizing it for all sorts of connections.

This is my comment from #2339 

https://github.com/redis/jedis/blob/af6ec7d2c705968f946804e9dec19eb2c4098073/src/main/java/redis/clients/jedis/JedisSentinelPool.java#L342-L344